src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -2297,6 +2297,18 @@ private BoundExpression BindNameofOperatorInternal(InvocationExpressionSyntax no
             {
                 diagnostics.AddAssembliesUsedByNamespaceReference(nsExpr.NamespaceSymbol);
             }
+            else if (boundArgument is BoundIndexerAccess indexerAccess)
+            {
+                // nameof(x[y]) where the argument is an indexer is an error. Ref analysis requires
+                // the indexer to be recognized as get, set, or both. We don't know which was
+                // intended for nameof() but since this is already an error, assume it's get.
+                Debug.Assert(!syntaxIsOk);
+                Debug.Assert(indexerAccess.AccessorKind == AccessorKind.Unknown);
+                if (indexerAccess.AccessorKind == AccessorKind.Unknown)
+                {
+                    boundArgument = CheckValue(boundArgument, BindValueKind.RValue, diagnostics);
```

Comment:
>diagnostics [](http://example.com/codeflow?start=84&length=11)

Consider passing ```BindingDiagnosticBag.Discarded``` to avoid unnecessary noise in diagnostics